onsite/lucxor/cli.py

```
@@ -1,781 +1,852 @@
+                if 'scan=' in native_id:
+                    scan_num = int(native_id.split('scan=')[-1])
+                    spectrum_map[scan_num] = spectrum
+                else:
+                    # Try to extract scan number from other formats
+                    for part in native_id.split():
+                        if part.isdigit():
+                            scan_num = int(part)
+                            spectrum_map[scan_num] = spectrum
+                            break
+            except Exception as e:
+                self.logger.warning(f"Cannot extract scan number from native ID: {native_id}, error: {str(e)}")
```

Ruff (0.14.0)

Use explicit conversion flag

Replace with conversion flag

(RUF010)